src/overlays/gamestates/ovl_select/z_select.h

```diff
@@ -0,0 +1,258 @@
+#ifndef Z_SELECT_H
```

Comment:
Shouldn't this file also include the struct for MapSelectState?

Reply:
iirc the headers make this impossible, I can't remember what's the real reason though, but I remember trying that and having issues

Reply:
It shouldn't be "impossible" but with how weird all the includes are right now in the repo we can focus on this later instead.

Reply:
I think the only issue is getting the gamestate struct without having looping includes, otherwise it should be fine